### ISSUE TITLE
Unexpected result after applying braces fixer for «else if» control structure

### ISSUE THREAD

**Issue by 27cm** [FriendsOfPHP/PHP-CS-Fixer#4148]
PHP 7.1.20
PHP CS Fixer 2.13.0

Config:
```php
'braces' => [
    'allow_single_line_closure'                   => true,
    'position_after_anonymous_constructs'         => 'same',
    'position_after_control_structures'           => 'same',
    'position_after_functions_and_oop_constructs' => 'same',
],
```

Sample input:
```php
if ($a) {
    // ...
} else if ($b) {
    // ...
} else {
    // ...
}
```

Actual fixer result:
```php
if ($a) {
    // ...
} else {
    if ($b) {
        // ...
    } else {
        // ...
    }
}
```

Expected: not changes.

**Comment by codigoergosum** [comment]
Agreed this is current behavior.  But it's unhelpful behavior: in the world of human brains, this issue describes a natural desire.  Failing initially to look through closed issues, it took me a couple hours to stop thinking of this behavior as a bug in the elseif fixer, that somehow setting elseif=>false did this thing and understanding it was a result of a naive braces rule.  Why not enhance braces rather requiring users to accept elseif?

**Comment by SpacePossum** [comment]
Hi and thanks for sharing your thoughts and experiences on this topic!

Currently the braces-fixer is being refactored to provide more easy configuration and far less responsibilities for that one rule. I hope this is a way forward suitable for you. If you can please review https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4884
